processing/src/main/java/io/druid/query/groupby/GroupByQueryEngine.java

```diff
@@ -148,7 +148,8 @@ public void close() throws IOException
     private final BufferAggregator[] aggregators;
     private final PositionMaintainer positionMaintainer;
 
-    private final TreeMap<ByteBuffer, Integer> positions;
+    private final Map<ByteBuffer, Integer> positions = Maps.newTreeMap();
```

Comment:
is there a reason to keep the treeMap at all? If ordering is required, would a linkedhashmap be sufficient?